src/Google.Api.Gax/Api/Gax/RetrySettings.cs

```diff
@@ -0,0 +1,156 @@
+        /// <param name="paramName">The parameter name to use in any exception thrown.</param>
+        /// <exception cref="ArgumentException">The settings are invalid.</exception>
+        public RetrySettings Validate(string paramName)
+        {
+            GaxPreconditions.CheckArgument(RetryBackoff != null, paramName, "No retry backoff specified");
+            GaxPreconditions.CheckArgument(TimeoutBackoff != null, paramName, "No timeout backoff specified");
+            GaxPreconditions.CheckArgument(DelayJitter != null, paramName, "No delay jitter specified");
+            GaxPreconditions.CheckArgument(RetryFilter != null, paramName, "No retry filter specified");
+            return this;
+        }
+
+        /// <summary>
+        /// Provides a mechanism for applying jitter to delays between retries.
+        /// See the <see cref="DelayJitter"/> property for more information.
+        /// </summary>
+        public interface IJitter
+        {
+            /// <summary>
+            /// Returns the actual delay to use given a maximum available delay.
+            /// </summary>
+            /// <param name="maxDelay">The maximum delay provided by the backoff settings</param>
+            /// <returns>The delay to use before retrying.</returns>
+            TimeSpan GetDelay(TimeSpan maxDelay);
+        }
+
+        /// <summary>
+        /// The default jitter which returns a uniformly distributed random delay between 0 and
+        /// the specified maximum.
+        /// </summary>
+        public static IJitter RandomJitter { get; } = new RandomJitterImpl();
+
+        /// <summary>
+        /// A jitter which simply returns the specified maximum delay.
+        /// </summary>
+        public static IJitter NoJitter { get; } = new NoJitterImpl();
+
+        // TODO: Is this a reasonable default? Does it make sense to have a default at all?
+
+        /// <summary>
+        /// The default retry filter, which retries operations which fail due to a status code of "not found".
+        /// </summary>
+        public static Predicate<RpcException> DefaultFilter { get; } = FilterForStatusCodes(StatusCode.NotFound);
+
+        /// <summary>
+        /// Creates a retry filter based on status codes.
+        /// </summary>
+        /// <param name="statusCodes">The status codes to retry. Must not be null.</param>
+        /// <returns>A retry filter based on status codes.</returns>
+        public static Predicate<RpcException> FilterForStatusCodes(params StatusCode[] statusCodes) =>
+            FilterForStatusCodes((IEnumerable<StatusCode>) statusCodes);
+
+        /// <summary>
+        /// Creates a retry filter based on status codes.
+        /// </summary>
+        /// <param name="statusCodes">The status codes to retry. Must not be null.</param>
+        /// <returns>A retry filter based on status codes.</returns>
+        public static Predicate<RpcException> FilterForStatusCodes(IEnumerable<StatusCode> statusCodes)
```

Comment:
It doesn't, and in particular I'd expect the set to be really small (1-3 codes), so just a linear search should be fine.